editor/src/clj/editor/scene_selection.clj

```diff
@@ -90,15 +82,26 @@
 (defn- select [controller op-seq mode toggle?]
   (let [select-fn (g/node-value controller :select-fn)
         selection (g/node-value controller :picking-selection)
+        contextual? (g/node-value controller :contextual?)
         mode-filter-fn (case mode
                         :single (fn [selection] (if-let [sel (first selection)] [sel] []))
-                        :multi  identity)
-        toggle-filter-fn (if toggle?
+                        :multi identity)
+        toggle-filter-fn (cond
+                           toggle?
                            (fn [selection]
                              (let [selection-set (set selection)
                                    prev-selection (g/node-value controller :prev-selection)
                                    prev-selection-set (set prev-selection)]
                                (into [] (concat (filter (complement selection-set) prev-selection) (filter (complement prev-selection-set) selection)))))
+
+                           contextual?
+                           (fn [selection]
+                             (let [prev-selection (g/node-value controller :prev-selection)]
+                               (if (some #(= (first selection) %) prev-selection)
+                                 prev-selection
+                                 selection)))
```

Comment:
If the element under the mouse is already selected, we should maintain the previous selection to avoid deselecting other elements.